d3/js/AliTV.js

```diff
@@ -959,6 +952,217 @@ AliTV.prototype.getOuterRadius = function() {
+};
+
+/**
+ * This method should call other filter functions in order to filter the visible chromosomes.
+ * @returns visibleChromosomes: returns only chromosomes which are visible
+ * @author Sonja Hohlfeld
+ */
+AliTV.prototype.filterChromosomes = function() {
+	var visibleChromosomes = this.data.karyo.chromosomes;
+	if (this.filters.showAllChromosomes === false) {
+		visibleChromosomes = this.filterVisibleChromosomes(visibleChromosomes);
+	}
+	if (this.filters.skipChromosomesWithoutLinks === true) {
+		visibleChromosomes = this.filterChromosomeWithoutLinkageInformation(visibleChromosomes);
+	}
+	if (this.filters.skipChromosomesWithoutVisibleLinks === true) {
+		visibleChromosomes = this.filterChromosomeWithoutVisibleLinks(visibleChromosomes);
+	}
+	return visibleChromosomes;
+};
+
+/**
+ *This method should filter all chromosome which are set visible in conf.filters.karyo.chromosomes[<chromosome>].visible
+ * @param visibleChromosomes: the method gets all current visible chromosomes.
+ * @returns filteredChromosomes: the method returns only chromosomes whose visibility is set true
+ * @author Sonja Hohlfeld 
+ */
+AliTV.prototype.filterVisibleChromosomes = function(visibleChromosomes) {
+	var that = this;
+	var filteredChromosomes = {};
+	$.each(visibleChromosomes, function(key, value) {
+		if (that.filters.karyo.chromosomes[key].visible === true) {
+			filteredChromosomes[key] = value;
+		}
+	});
+	return filteredChromosomes;
+};
+
+/**
+ *This method should filter all chromosome which have no linkage information
+ * @param visibleChromosomes: the method gets all current visible chromosomes.
+ * @returns filteredChromosomes: the method returns only chromosomes which have linkage information
+ * @author Sonja Hohlfeld 
+ */
+AliTV.prototype.filterChromosomeWithoutLinkageInformation = function(visibleChromosomes) {
+	var that = this;
+	var filteredChromosomes = {};
+	$.each(visibleChromosomes, function(key, value) {
+		var currentChromosome = key;
+		var valueOfCurrentChromosome = value;
+		$.each(that.data.links, function(key, value) {
+			if (that.data.features[value.source].karyo === currentChromosome && (currentChromosome in filteredChromosomes) === false || that.data.features[value.target].karyo === currentChromosome && (currentChromosome in filteredChromosomes) === false) {
+				filteredChromosomes[currentChromosome] = valueOfCurrentChromosome;
+			}
+		});
+	});
+	return filteredChromosomes;
+};
+
+/**
+ * This method should filter all chromosome which have no visible links with the current configurations
+ * @param visibleChromosomes: the method gets all current visible chromosomes.
+ * @returns filteredChromosomes: the method returns only chromosomes which have visible links
+ * @author Sonja Hohlfeld 
+ */
+AliTV.prototype.filterChromosomeWithoutVisibleLinks = function(visibleChromosomes) {
+	var that = this;
+	var filteredChromosomes = {};
+	var allLinks = that.data.links;
+	var filteredLinks = that.filterLinksByIdentity(allLinks);
```

Comment:
Ok, as pointed out by @sonjahohlfeld that would currently lead to an infinite loop as ```filterLinks()``` calls ```filterChromosomes()``` and reverse. The best way to avoid this dilemma is to pass the ```filteredChromosomes``` as a parameter to ```filterLinks()``` and do not call ```filterChromosomes``` from within.
```{javascript}
AliTV.prototype.filterLinks = function(filteredChromosomes){
...
}
```